server/main.py

```diff
@@ -139,10 +143,38 @@ async def broadcast(self, data: dict, room_name: str, sender: Client | None = No
                 continue
             await connection.send(data)
 
+    def room_exists(self, room_code: str) -> bool:
+        """Checks if a room exists.
+
+        Args:
+            room_code: The code associated with a particular room.
+
+        Returns:
+            True fi the room exists. False otherwise.
+        """
+        if room_code in self._rooms:
+            return True
+        return False
+
 
 manager = ConnectionManager()
 
 
+class ConnectionEventData(BaseModel):
+    """A model representing the connection event data."""
+
+    message: str
+    difficulty: int
+    room_code: str
+
+
+class ConnectionData(BaseModel):
+    """A model representing the initial connection."""
+
+    type: str
```

Comment:
Since the `type` key is required in every request, we could make a `BaseModel` called `Event` that has a field `type` and a field `data`. The field `type` should be an `Enum` variant, while the `data` should be an instance of something like `ConnectionData`, we can then create different classes to represent the data of different types of events.

Something like this:
```py
class EventType(Enum):
    ...

class EventData(BaseModel):
    pass

class ConnectionData(EventData):
    message: str
    difficulty: int
    room_code: str

class InsertData(EventData):
    ...

class Event:
    type: EventType
    data: EventData
```

Reply:
Yup. Was going to create another PR for this and have these models in a different file

Reply:
I agree with that, this looks like it is going to be a big part of the project.

Reply:
#31